### DISCUSSION TITLE
import 的时候能默认从 es包里引么

### DISCUSSION THREAD

**Discussion by Jackson-p** [ant-design/ant-design#30047] [Q&A]
因为要遵循esm规范，所以需要用到antd的es包
![image](https://user-images.githubusercontent.com/20417123/113855525-b7f2c880-97d2-11eb-9b83-0957481c87b1.png)
但是command点击进入定义之后发现其实antd默认引的都是lib，有办法换成es么
![image](https://user-images.githubusercontent.com/20417123/113855722-f1c3cf00-97d2-11eb-9973-29450a7170fd.png)

**Comment by yoyo837** [comment]
这由npm package.json 的main 字段决定

**Reply by Jackson-p** [reply]
对的 antd 的package.json里面就是既有main又有module，main是lib，所以问题就是怎样去默认读取es呀😂

**Reply by yoyo837** [reply]
这个是编辑器有关吧

**Reply by Jackson-p** [reply]
好吧 感谢

**Reply by chentsulin** [reply]
未來可能可以靠 import conditions 解決這個問題

https://www.typescriptlang.org/docs/handbook/esm-node.html

**Comment by yoyo837** [comment]
bundle工具可能附加有别的逻辑

**Reply by tangdw** [reply]
antd-mobile 的 es 目录下还有 require 写法，vite 表示不支持。这可咋整

![image](https://user-images.githubusercontent.com/17575065/114716935-2271ae80-9d67-11eb-83af-8d2af1f5efb1.png)
